test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -78,6 +78,7 @@
 public abstract class ESSingleNodeTestCase extends ESTestCase {
 
     private static Node NODE = null;
+    private final Path nodeDir = createTempDir();
```

Comment:
I'm not sure why this change is needed.
Given that the only caller of `getNodeDir` is overriding node settings, can't you just create two temporary directories there?

Reply:
I thought we could centralize the node directory. I created another temporary directories instead as you suggested.